### PR TITLE
Migrated `instancenetworking` datasource to Framework

### PR DESCRIPTION
**Pull request by ezilber-akamai**
## 📝 Description

Migrated `instancenetworking` datasource to Framework

## ✔️ How to Test

`make PKG_NAME="linode/instancenetworking" testacc`

Ticket: TPT-1983

### CODE REVIEW

**Comment by ezilber-akamai** [comment]
/acctest sha=7656432407ead1db16591a1ca5394ed00e1f1e94 pkg=linode/instancenetworking

File: linode/instancenetworking/flatten.go
Comment (yec-akamai) [inline linode/instancenetworking/flatten.go:1]: Probably better to rename this file to be `framework_models.go`, and move `parseInstanceIPAddressResponse` and `DataSourceModel` into it as well. So we can have one place focusing on parsing the data model and simplify the datasource file.

**Comment by ezilber-akamai** [comment]
/acctest sha=6aa1a1d5571d090717d04d43a6a4092767462c3f pkg=linode/instancenetworking

**Comment by ezilber-akamai** [comment]
/acctest sha=6ec92ecebf6b5088679660d32e0e41b646067a77 pkg=linode/instancenetworking

**Review by lgarber-akamai — APPROVED** [review]
Nice work!

File: linode/instancenetworking/framework_datasource.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to get Instance Networking Information: ", err.Error(),
+		)
+		return
+	}
+
+	data.parseInstanceIPAddressResponse(ctx, netInfo)
+
+	resp.Diagnostics.Append(data.parseInstanceIPAddressResponse(ctx, netInfo)...)
```
Comment (zliang-akamai) [inline linode/instancenetworking/framework_datasource.go:77]:
```suggestion
	resp.Diagnostics.Append(data.parseInstanceIPAddressResponse(ctx, netInfo)...)
```

I think you would only need to call `data.parseInstanceIPAddressResponse` once. Or is there any reason to call it twice?

**Review by yec-akamai — APPROVED** [review]
Looks good!

File: linode/instancenetworking/framework_datasource_schema.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	AttrTypes: map[string]attr.Type{
+		"address":     types.StringType,
+		"gateway":     types.StringType,
+		"prefix":      types.Int64Type,
+		"rdns":        types.StringType,
+		"region":      types.StringType,
+		"subnet_mask": types.StringType,
+		"type":        types.StringType,
+	},
+}
```
Comment (zliang-akamai) [inline linode/instancenetworking/framework_datasource_schema.go:19]: In the latest API, the response include an extra attribute `public`. Although we have the key values to show whether it's public (e.g., `ipv4.public` only contains public IP and `ipv4.private` only contains private IP), can we also include it here for user's convenient?

File: linode/instancenetworking/framework_datasource.go (hunk 1 above)
Comment (ezilber-akamai) [reply linode/instancenetworking/framework_datasource.go:77]: You're right. Just a copy-paste mistake. I will fix this!

**Comment by zliang-akamai** [comment]
Others look good! Nice work!

File: linode/instancenetworking/framework_datasource_schema.go (hunk 2 above)
Comment (ezilber-akamai) [reply linode/instancenetworking/framework_datasource_schema.go]: I don't see an issue with keeping the datasource schema aligned with the current state of the API. I also noticed that I was missing `LinodeID` so I added that as well. Does anyone else have any thoughts?

**Comment by ezilber-akamai** [comment]
/acctest sha=1e48fcd45f6310a7ab1d03ebd4c77ac7094a90b2 pkg=linode/instancenetworking

**Review by zliang-akamai — APPROVED** [review]

**Pull request merged**